### ISSUE TITLE
Flicker/Sound glitches when using remote play with Boxtron

### ISSUE THREAD

**Issue by Pipeline79** [dreamer/boxtron#45]
This is a bit of a complex issue that I will do my best to describe clearly.
I'm using a laptop with an external gpu, running a dosbox game in steam with boxtron.  I'm casting the game to my TV with the steam remote play feature and a steam link.
The game runs fine if I were only to play it on my monitor attached to the gpu.
With egpus it's sometimes recommended to use only the external display and not also render the game on the laptop screen as well, for better performance.
I am finding however, that if I switch off the laptop display in my desktop settings this results in continuous flickering in the cast display of the game on my TV - but not on the monitor attached to the computer/gpu itself.  It seems to be caused by my moving the mouse cursor - perhaps related to the steam overlay.  If I switch on my laptop screen again, the flickering on the TV stops.
In this way there is instead a few sound/lag glitches, but it is better.
I notice that when the screen is open like this, and I look at the computer, the 'window' of the dosbox game is in fullscreen on the external monitor, but there is an overlap with the edge of the game window extending onto the laptop display as well.

I'm posting this here because I get the flickering problems only when running dosbox games in Steam and casting them, not with other Steam games and casting them.  Even though I would expect modern games to be more demanding to play and cast from my computer.

Perhaps someone here knows what this means and if there's anything I can change to correct this?  Changing dosbox/boxtron configuration file somehow? 

Thank you.

**Comment by dreamer** [comment]
Do you use DOSBox or DOSBox Staging?

**Comment by Pipeline79** [comment]
Hi, I was using the standard dosbox that came with Ubuntu at first, then I switched to their dosbox-daily from a different ppa.
It says it's dosbox 0.74-5-3
They both have the same problems I described above.

**Comment by dreamer** [comment]
0.74.x (any version) is based on 10-year old, outdated, broken, code - I really can't do anything about it :(

Try dosbox-staging, please (latest release is 0.75.2 ATM, we will make 0.76.0 release very soon).

**Comment by Pipeline79** [comment]
Ok, i will give it a go.  Thank you.

**Comment by Pipeline79** [comment]
I've tried dosbox-staging, but sadly that gives me a completely black screen whether the laptop display is on or not.  I can hear sound, and the controls respond.  It goes black just after the 'DosBox Staging' logo comes up and the dosbox command window finishes.  It appears to work fine on the computer itself.
Is remote play with boxtron something that is known to work for other people?

My external monitor is 1920x1080p at 60Hz, my laptop is 1600x900 at 60Hz.
My TV is capable of 1920x1080i and 59.94Hz, but streaming is set to 720p. (Old TV).
Anything to do with that?  Doesn't seem to matter with other games.

**Comment by dreamer** [comment]
I tested remote play only once, very long time ago.

So you see the splash screen and then black screen? Interesting… Thank you for the report, I will look into it :)

**Comment by Pipeline79** [comment]
Yeah, I see the splash screen, the dosbox messages, but as soon as the games graphics start it goes black.
However, I wonder if this is my hardware setup with a laptop and an egpu over expresscard that is the problem, not you.
I tried it again just now and when remote play is active both screens go black - on the TV and monitor.
What makes it confusing is that if I use the old windows dosbox with the game and disable boxtron in Steam I can stream OK.
I get the visual glitch that you mentioned here https://github.com/dreamer/boxtron/issues/8 with it instead, etc.

**Comment by dreamer** [comment]
No, I think it's rather something with DOSBox Staging implementation. I don't experience this problem exactly, but when using MangoHud it disappears (also right after splash screen). So perhaps one of fixes I implemented is causing an unexpected side effect.
